angular/projects/public-lng/src/app/explore-panel/explore-panel.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit, Output, EventEmitter, Input, ChangeDetectorRef } from '@angular/core';
+
+import { FilterSection } from '../models/document-filter';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-explore-panel',
+  templateUrl: './explore-panel.component.html',
+  styleUrls: ['./explore-panel.component.scss']
+})
+export class ExplorePanelComponent implements OnInit {
+  @Input() filterSections: FilterSection[] = []; // document filter sections // used in template
+
+  @Output() updateFilters = new EventEmitter();
+  @Output() hideSidePanel = new EventEmitter();
+
+  readonly minDate = new Date('01-01-1900'); // first app created
+  readonly maxDate = new Date(); // today
+
+  public _dateRangeFromFilter: Date = null;
+  public _dateRangeToFilter: Date   = null;
+
+  public textFilterKeys: any[];
+  public filter = [];
+
+  constructor(
+    private _changeDetectionRef: ChangeDetectorRef,
+    private route: ActivatedRoute
+    ) {
+      this.textFilterKeys = [];
+    }
+
+  public ngOnInit() {
+    // For each filter coming in, make the filter bucket array
+    this.route.params.subscribe(params => {
```

Comment:
I'm not actually clear on how important ```takeUntil(this.ngUnsubscribe)``` is, but something that could be added if needed.

Reply:
It's an error, will fix.